src/components/dialogs/commons/use-name-check.tsx

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import React, { ReactElement, useCallback, useEffect, useState } from 'react';
+import { useIntl } from 'react-intl';
+import { CircularProgress, InputAdornment } from '@mui/material';
+import { elementExists } from '../../../utils/rest-api';
+import CheckIcon from '@mui/icons-material/Check';
+import { useSelector } from 'react-redux';
+import { useDebounce } from '@gridsuite/commons-ui';
+import { FieldError } from 'react-hook-form';
+import { ElementType } from '../../../utils/elementType';
+import { ReduxState } from '../../../redux/reducer.type';
+
+export type NameCheckReturn = [ReactElement | null, boolean];
+
+interface INameCheckProps<T> {
+    field: keyof T;
+    value: string;
+    elementType: ElementType;
+    setError: (field: keyof T, error: FieldError) => void;
+}
+
+export const useNameCheck = <T extends Record<string, any>>({
+    field,
+    value,
```

Comment:
as discussed, we can not remove field and value in use-name-check, because we need isChecking returned from this custom hook..
so useNameCheck will be outside the formProvider of CustomMuiDialog..
I will use a async checking in the form schema in a next TS. this way we will not use useNameCheck